tableauserverclient/server/endpoint/endpoint.py

```diff
@@ -69,3 +70,33 @@ def post_request(self, url, xml_request, content_type='text/xml'):
                                   content=xml_request,
                                   auth_token=self.parent_srv.auth_token,
                                   content_type=content_type)
+
+
+def api(version):
+    '''Annotate the minimum supported version for an endpoint.
+
+    Checks the version on the server object and does a float comparison.
+    It will raise an exception if the server version is > the version specified.
+
+    Args:
+        `version` minimum version that supports the endpoint. Float.
+    Raises:
+        EndpointUnavailableError
+    Returns:
+        None
+
+    Example:
+    >>> @api(version=2.3)
+    >>> def get(self, req_options=None):
+    >>>     ...
+    '''
+    def _decorator(func):
+        @wraps(func)
+        def wrapper(self, *args, **kwargs):
+            server_version = self.parent_srv.version
+            if float(server_version) < version:
```

Comment:
Please use https://docs.python.org/3.6/distutils/apiref.html#module-distutils.version to compare version numbers.  Casting to float will break if/when we decide to be more nuanced with your version numbers.